File: apps/polyglot-sample/sample.php
<?php

function answer(): int {
    return 42;
}

function main(): void {
}

class Counter {
    private int $value = 0;

    function __construct(int $start) {
        $this->value = $start;
    }

    function inc(): int {
        $v = $this->value + 1;
        $this->value = $v;
        return $v;
    }
}

interface Drawable {
    function draw(): string;
}

File: apps/polyglot-sample/sample.pl
sub answer {
    return 42;
}

sub main {
    my $value = answer();
    return;
}

package Counter;
sub new {
    my ($class, $start) = @_;
    my $self = {
        value => $start,
    };
    bless $self, $class;
    return $self;
}
sub inc {
    my ($self) = @_;
    $self->{value} += 1;
    return $self->{value};
}
